spec/validation/configure_spec.rb

```diff
@@ -34,244 +41,264 @@
                'and node will cause an error.',
       },
 
-      domain: {
-        string_question: {
+      domain: [
+        {
+          identifier: 'string_question',
           question: 'Am I a string question without a default and type?',
         },
-        integer_question: {
+        {
+          identifier: 'integer_question',
           question: 'Am I an integer question with a default?',
           type: 'integer',
           default: 10,
         },
-        boolean_true: {
+        {
+          identifier: 'boolean_true',
           question: 'Can I have a boolean true (/yes) default?',
           type: 'boolean',
-          default: 'yes',
+          default: true,
```

Comment:
In the commit message (https://github.com/alces-software/metalware/pull/185/commits/8ff5ded96c5a57c12fe0d5f5734397d9e75d60af) you say:

> The Configurator will need to accomidate it when it is refactored

Unless I'm missing what you mean, I think the Configurator should just work if you pass it a `boolean` question with a Boolean `default`; I believe it currently works in this case. I don't think the preceding change would have worked (requiring `'yes'`/`'no'` defaults), but you've removed that so this should be fine now (probably :slightly_smiling_face: ).

Reply:
Will look into this when I refactor `Configurator`. I am really not sure what will happen. But as far as the validation goes, this is the correct behaviour.